types/index.d.ts

```diff
@@ -266,17 +395,105 @@ declare namespace unified {
    * @typeParam S Plugin settings
    * @typeParam P Processor settings
    */
-  type Pluggable<S extends any[] = [Settings?], P = Settings> =
-    | Plugin<S, P>
-    | Preset<S, P>
-    | PluginTuple<S, P>
+  type Pluggable<
+    P = Settings,
+    S1 extends any[] = [Settings?],
+    S2 extends any[] = [Settings?],
+    S3 extends any[] = [Settings?],
+    S4 extends any[] = [Settings?],
+    S5 extends any[] = [Settings?],
+    S6 extends any[] = [Settings?],
+    S7 extends any[] = [Settings?],
+    S8 extends any[] = [Settings?],
+    S9 extends any[] = [Settings?],
+    S10 extends any[] = [Settings?],
+    S11 extends any[] = [Settings?],
+    S12 extends any[] = [Settings?],
+    S13 extends any[] = [Settings?],
+    S14 extends any[] = [Settings?],
+    S15 extends any[] = [Settings?],
+    S16 extends any[] = [Settings?],
+    S17 extends any[] = [Settings?],
+    S18 extends any[] = [Settings?],
+    S19 extends any[] = [Settings?],
+    S20 extends any[] = [Settings?]
+  > =
+    | Plugin<P, S1>
+    | PluginTuple<P, S1>
+    | Preset<
+        P,
+        S1,
+        S2,
+        S3,
+        S4,
+        S5,
+        S6,
+        S7,
+        S8,
+        S9,
+        S10,
+        S11,
+        S12,
+        S13,
+        S14,
+        S15,
+        S16,
+        S17,
+        S18,
+        S19,
+        S20
+      >
 
   /**
    * A list of plugins and presets
    *
    * @typeParam P Processor settings
    */
-  type PluggableList<P = Settings> = Array<Pluggable<[any?], P>>
+  type PluggableList<
+    P = Settings,
+    S1 extends any[] = [Settings?],
+    S2 extends any[] = [Settings?],
+    S3 extends any[] = [Settings?],
+    S4 extends any[] = [Settings?],
+    S5 extends any[] = [Settings?],
+    S6 extends any[] = [Settings?],
+    S7 extends any[] = [Settings?],
+    S8 extends any[] = [Settings?],
+    S9 extends any[] = [Settings?],
+    S10 extends any[] = [Settings?],
+    S11 extends any[] = [Settings?],
+    S12 extends any[] = [Settings?],
+    S13 extends any[] = [Settings?],
+    S14 extends any[] = [Settings?],
+    S15 extends any[] = [Settings?],
+    S16 extends any[] = [Settings?],
+    S17 extends any[] = [Settings?],
+    S18 extends any[] = [Settings?],
+    S19 extends any[] = [Settings?],
+    S20 extends any[] = [Settings?]
+  > = [
+    Pluggable<P, S1>?,
+    Pluggable<P, S2>?,
+    Pluggable<P, S3>?,
+    Pluggable<P, S4>?,
+    Pluggable<P, S5>?,
+    Pluggable<P, S6>?,
+    Pluggable<P, S7>?,
+    Pluggable<P, S8>?,
+    Pluggable<P, S9>?,
+    Pluggable<P, S10>?,
+    Pluggable<P, S11>?,
+    Pluggable<P, S12>?,
+    Pluggable<P, S13>?,
+    Pluggable<P, S14>?,
+    Pluggable<P, S15>?,
+    Pluggable<P, S16>?,
+    Pluggable<P, S17>?,
+    Pluggable<P, S18>?,
+    Pluggable<P, S19>?,
+    Pluggable<P, S20>?,
+    ...Array<Pluggable<P, any[]>>
```

Comment:
allows more plugins to be added past 20, but without enhanced settings type checking